cime_config/machines/cmake_macros/intel.cmake

```diff
@@ -27,7 +27,7 @@ if (compile_threaded)
   string(APPEND FFLAGS " -qopenmp")
 endif()
 if (DEBUG)
-  string(APPEND FFLAGS " -O0 -g -check uninit -check bounds -check pointers -fpe0 -check noarg_temp_created")
+  string(APPEND FFLAGS " -O0 -g -check uninit -check bounds -check pointers -check noarg_temp_created")
```

Comment:
Just noticed this -- why is this PR changing flags?
Same about intel_compy.cmake .

Reply:
Thanks for looking at this closely @oksanaguba.  The flag changes for Compy were needed to run the model in DEBUG mode (courtesy of @peterdschwartz).  Without these changes, I could not test my results in DEBUG mode.  These changes aren't needed for checking carbon conservation in EAM, so if it makes sense, we could separate these issues into a different PR.  There was also a bug found in the land portion (again by @peterdschwartz) that was causing issues in DEBUG mode.

Reply:
Thanks @ndkeen I agree. See also that flags are changed in config_machines.

Reply:
Well as i describe in the commit, the floating point exception debug flags hinder debugging for ELM due to the NaN logic in the code, so I don't think they should be turned on by default at least.   
To allow other components to use it, then I could put a if statement to check the component name inside the DEBUG if statment.

Reply:
@peterdschwartz what does 'hinder' mean in this case? Would you please tell which scenario we are dealing with here:
```
Sorry I am not very clear, but it seems to me that the cases are:

(unlikely) ELM needs changes in DEBUG flags to run from master. Make a new small PR, someone from performance should approve it. Remove change of flags from this PR. Merge flags PR first, before this PR.

only your PR needs new DEBUG flags. Let's figure out why (since it is just 1 line of code) and fix the code instead of flags. Revert flags in this PR back.

your PR does not need these flags anymore. Remove them then.
```

Reply:
@oksanaguba You are mistaken about @beharrop initially modifying elm files in this PR.  I was only able to fix that bug after removing the debug flags in question and CNPBudgetMod.F90 appears to have been last modified in master in Nov.  So that's what I mean by  it hindering debugging.  Moreover, I just reran a test case on compy after re-adding the `-fpe0` flag and can confirm that it fails again. 
I agree that these floating point exceptions raise questions about the consistencies of use of NaNs with ELM as I have noticed this issue several times in the past couple years which is why I knew how to fix it.  I think that not compiling elm files with fpe0 or similar flags is a good choice in the short term.

Reply:
@peterdschwartz I refreshed browser and still saw 1 elm file that this PR touched, components/elm/src/biogeochem/CNPBudgetMod.F90 , and it changes DEBUG flags. It is true I do not understand what's going on. Would you please tell me which case we are dealing with here, see below. Sorry that I keep asking this, but the list below (it seems to me) covers all possible cases why the flag changes are in this PR.

1.(unlikely) ELM needs changes in DEBUG flags to run from master. Make a new small PR, someone from performance should approve it. Remove change of flags from this PR. Merge flags PR first, before this PR.

2.only your PR needs new DEBUG flags. Let's figure out why (since it is just 1 line of code) and fix the code instead of flags. Revert flags in this PR back.

3.your PR does not need these flags anymore. Remove them then.

Reply:
> @peterdschwartz I refreshed browser and still saw 1 elm file that this PR touched, components/elm/src/biogeochem/CNPBudgetMod.F90 , and it changes DEBUG flags. It is true I do not understand what's going on.

In my previous comment, I was trying to clarify the timeline of commits.  With your cases listed, then I say it is Case 1 -- I don't know why you think it's unlikely as there are multiple configs and gaps in testing. 
1. @beharrop  experienced fails in DEBUG mode **without** changing any elm files.  
2. The debug mode fails pointed to lines of code not related to CNPBudgetMod.F90 
3. Me removing the related debug flags resulted in a much more informative backtrace that showed there was a bug in CNPBudgetMod.F90 which was last changed in **master** in Nov.
4. Adding back -fpe0 flag **after** fixing the typo in CNPBudgetMod.f90 still results in fails.
5. The fails are not due to this PR and maybe should not even be considered bugs since ELM relies on NaNs always evaluating to **false** in certain parts of the code.  It's more of a question of coding paradigm here i think.

If this does not clarify, I can try to talk with you next week sometime.

Reply:
Thanks! Then flags should go into a separate PR, and as you suggested, they should be safeguarded by an if-statement for elm. Though I do not know if that works for files like intel.cmake, etc. 

Would you be willing to make such PR?

Another option is to remove flags from here and leave debug flags in master as is. There are _D cases in tests, but they do not trigger this issue. This is another reason to document, what exactly was run for this PR.